#!/bin/bash -l

#$ -e /dev/null
#$ -o /dev/null
#$ -P tuberculosis
#$ -pe omp 16
#$ -t 1-84

module load sratoolkit/2.10.5
module load nextflow/19.10.0

INDEX=$(($SGE_TASK_ID-1))

INPUT=(SRR9163023 SRR9163024 SRR9163025 SRR9163026 SRR9163027 SRR9163028 SRR9163029 SRR9163030 SRR9163031 SRR9163032 SRR9163033 SRR9163034 SRR9163035 SRR9163036 SRR9163037 SRR9163038 SRR9163039 SRR9163040 SRR9163041 SRR9163042 SRR9163043 SRR9163044 SRR9163045 SRR9163046 SRR9163047 SRR9163048 SRR9163049 SRR9163050 SRR9163051 SRR9163052 SRR9163053 SRR9163054 SRR9163055 SRR9163056 SRR9163057 SRR9163058 SRR9163059 SRR9163060 SRR9163061 SRR9163062 SRR9163063 SRR9163064 SRR9163065 SRR9163066 SRR9163067 SRR9163068 SRR9163069 SRR9163070 SRR9163071 SRR9163072 SRR9163073 SRR9163074 SRR9163075 SRR9163076 SRR9163077 SRR9163078 SRR9163079 SRR9163080 SRR9163081 SRR9163082 SRR9163083 SRR9163084 SRR9163085 SRR9163086 SRR9163087 SRR9163088 SRR9163089 SRR9163090 SRR9163091 SRR9163092 SRR9163093 SRR9163094 SRR9163095 SRR9163096 SRR9163097 SRR9163098 SRR9163099 SRR9163100 SRR9163101 SRR9163102 SRR9163103 SRR9163104 SRR9163105 SRR9163106)
SRRID=${INPUT[$INDEX]}

SERIES_LIBRARY_LAYOUT=(SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE)
SAMPLE_LIBRARY_LAYOUT=${SERIES_LIBRARY_LAYOUT[$INDEX]}

if [[ -e "$SRRID.tsv.gz" ]]; then
  exit
fi

prefetch --output-directory "/scratch/$USER" $SRRID | grep "\S" &> "$SRRID.log"
fasterq-dump --outdir "/scratch/$USER/$SRRID" --temp "/scratch/$USER/$SRRID" --threads 16 "/scratch/$USER/$SRRID" &>> "$SRRID.log"

if [[ $SAMPLE_LIBRARY_LAYOUT == "PAIRED" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*{1,2}.fastq" --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ $SAMPLE_LIBRARY_LAYOUT == "SINGLE" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*.fastq" --singleEnd --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ -n $(grep -l "Succeeded   : 13" "$SRRID.log") ]]; then
  gzip --stdout "/scratch/$USER/$SRRID/results/featureCounts/merged_gene_counts.txt" > "$SRRID.tsv.gz"
fi

rm -rf "/scratch/$USER/$SRRID"

