#!/bin/bash -l

#$ -e /dev/null
#$ -o /dev/null
#$ -P tuberculosis
#$ -pe omp 16
#$ -t 1-138

module load sratoolkit/2.10.5
module load nextflow/19.10.0

INDEX=$(($SGE_TASK_ID-1))

INPUT=(SRR6370167 SRR6370168 SRR6370169 SRR6370170 SRR6370171 SRR6370172 SRR6370173 SRR6370174 SRR6370175 SRR6370176 SRR6370177 SRR6370178 SRR6370179 SRR6370180 SRR6370181 SRR6370182 SRR6370183 SRR6370184 SRR6370185 SRR6370186 SRR6370187 SRR6370188 SRR6370189 SRR6370190 SRR6370191 SRR6370192 SRR6370193 SRR6370194 SRR6370195 SRR6370196 SRR6370197 SRR6370198 SRR6370199 SRR6370200 SRR6370201 SRR6370202 SRR6370203 SRR6370204 SRR6370205 SRR6370206 SRR6370207 SRR6370208 SRR6370209 SRR6370210 SRR6370211 SRR6370212 SRR6370213 SRR6370214 SRR6370215 SRR6370216 SRR6370217 SRR6370218 SRR6370219 SRR6370220 SRR6370221 SRR6370222 SRR6370223 SRR6370224 SRR6370225 SRR6370226 SRR6370227 SRR6370228 SRR6370229 SRR6370230 SRR6370231 SRR6370232 SRR6370233 SRR6370234 SRR6370235 SRR6370236 SRR6370237 SRR6370238 SRR6370239 SRR6370240 SRR6370241 SRR6370242 SRR6370243 SRR6370244 SRR6370245 SRR6370246 SRR6370247 SRR6370248 SRR6370249 SRR6370250 SRR6370251 SRR6370252 SRR6370253 SRR6370254 SRR6370255 SRR6370256 SRR6370257 SRR6370258 SRR6370259 SRR6370260 SRR6370261 SRR6370262 SRR6370263 SRR6370264 SRR6370265 SRR6370266 SRR6370267 SRR6370268 SRR6370269 SRR6370270 SRR6370271 SRR6370272 SRR6370273 SRR6370274 SRR6370275 SRR6370276 SRR6370277 SRR6370278 SRR6370279 SRR6370280 SRR6370281 SRR6370282 SRR6370283 SRR6370284 SRR6370285 SRR6370286 SRR6370287 SRR6370288 SRR6370289 SRR6370290 SRR6370291 SRR6370292 SRR6370293 SRR6370294 SRR6370295 SRR6370296 SRR6370297 SRR6370298 SRR6370299 SRR6370300 SRR6370301 SRR6370302 SRR6370303 SRR6370304)
SRRID=${INPUT[$INDEX]}

SERIES_LIBRARY_LAYOUT=(PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED)
SAMPLE_LIBRARY_LAYOUT=${SERIES_LIBRARY_LAYOUT[$INDEX]}

if [[ -e "$SRRID.tsv.gz" ]]; then
  exit
fi

prefetch --output-directory "/scratch/$USER" $SRRID | grep "\S" &> "$SRRID.log"
fasterq-dump --outdir "/scratch/$USER/$SRRID" --temp "/scratch/$USER/$SRRID" --threads 16 "/scratch/$USER/$SRRID" &>> "$SRRID.log"

if [[ $SAMPLE_LIBRARY_LAYOUT == "PAIRED" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*{1,2}.fastq" --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ $SAMPLE_LIBRARY_LAYOUT == "SINGLE" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*.fastq" --singleEnd --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ -n $(grep -l "Succeeded   : 13" "$SRRID.log") ]]; then
  gzip --stdout "/scratch/$USER/$SRRID/results/featureCounts/merged_gene_counts.txt" > "$SRRID.tsv.gz"
fi

rm -rf "/scratch/$USER/$SRRID"

