#!/bin/bash -l

#$ -e /dev/null
#$ -o /dev/null
#$ -P tuberculosis
#$ -pe omp 16
#$ -t 1-84

module load sratoolkit/2.10.5
module load nextflow/19.10.0

INDEX=$(($SGE_TASK_ID-1))

INPUT=(SRR8197969 SRR8197970 SRR8197971 SRR8197972 SRR8197973 SRR8197974 SRR8197975 SRR8197976 SRR8197977 SRR8197978 SRR8197979 SRR8197980 SRR8197981 SRR8197982 SRR8197983 SRR8197984 SRR8197985 SRR8197986 SRR8197987 SRR8197988 SRR8197989 SRR8197990 SRR8197991 SRR8197992 SRR8197993 SRR8197994 SRR8197995 SRR8197996 SRR8197997 SRR8197998 SRR8197999 SRR8198000 SRR8198001 SRR8198002 SRR8198003 SRR8198004 SRR8198005 SRR8198006 SRR8198007 SRR8198008 SRR8198009 SRR8198010 SRR8198011 SRR8198012 SRR8198013 SRR8198014 SRR8198015 SRR8198016 SRR8198017 SRR8198018 SRR8198019 SRR8198020 SRR8198021 SRR8198022 SRR8198023 SRR8198024 SRR8198025 SRR8198026 SRR8198027 SRR8198028 SRR8198029 SRR8198030 SRR8198031 SRR8198032 SRR8198033 SRR8198034 SRR8198035 SRR8198036 SRR8198037 SRR8198038 SRR8198039 SRR8198040 SRR8198041 SRR8198042 SRR8198043 SRR8198044 SRR8198045 SRR8198046 SRR8198047 SRR8198048 SRR8198049 SRR8198050 SRR8198051 SRR8198052)
SRRID=${INPUT[$INDEX]}

SERIES_LIBRARY_LAYOUT=(SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE)
SAMPLE_LIBRARY_LAYOUT=${SERIES_LIBRARY_LAYOUT[$INDEX]}

if [[ -e "$SRRID.tsv.gz" ]]; then
  exit
fi

prefetch --output-directory "/scratch/$USER" $SRRID | grep "\S" &> "$SRRID.log"

if [[ -d $SRRID ]]; then
  mv $SRRID/* "/scratch/$USER/$SRRID" && rm -rf $SRRID
fi

(cd "/scratch/$USER" && fasterq-dump --outdir $SRRID --temp $SRRID --threads 16 $SRRID) &>> "$SRRID.log"

if [[ $SAMPLE_LIBRARY_LAYOUT == "PAIRED" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*{1,2}.fastq" --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ $SAMPLE_LIBRARY_LAYOUT == "SINGLE" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*.fastq" --singleEnd --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ -n $(grep -l "Succeeded   : 13" "$SRRID.log") ]]; then
  gzip --stdout "/scratch/$USER/$SRRID/results/featureCounts/merged_gene_counts.txt" > "$SRRID.tsv.gz"
fi

rm -rf "/scratch/$USER/$SRRID"

