#!/bin/bash -l

#$ -e /dev/null
#$ -o /dev/null
#$ -P tuberculosis
#$ -pe omp 16
#$ -t 1-100

module load sratoolkit/2.10.5
module load nextflow/19.10.0

INDEX=$(($SGE_TASK_ID-1))

INPUT=(SRR13571261 SRR13571262 SRR13571263 SRR13571264 SRR13571265 SRR13571266 SRR13571267 SRR13571268 SRR13571269 SRR13571270 SRR13571271 SRR13571272 SRR13571273 SRR13571274 SRR13571275 SRR13571276 SRR13571277 SRR13571278 SRR13571279 SRR13571280 SRR13571281 SRR13571282 SRR13571283 SRR13571284 SRR13571285 SRR13571286 SRR13571287 SRR13571288 SRR13571289 SRR13571290 SRR13571291 SRR13571292 SRR13571293 SRR13571294 SRR13571295 SRR13571296 SRR13571297 SRR13571298 SRR13571299 SRR13571300 SRR13571301 SRR13571302 SRR13571303 SRR13571304 SRR13571305 SRR13571306 SRR13571307 SRR13571308 SRR13571309 SRR13571310 SRR13571311 SRR13571312 SRR13571313 SRR13571314 SRR13571315 SRR13571316 SRR13571317 SRR13571318 SRR13571319 SRR13571320 SRR13571321 SRR13571322 SRR13571323 SRR13571324 SRR13571325 SRR13571326 SRR13571327 SRR13571328 SRR13571329 SRR13571330 SRR13571331 SRR13571332 SRR13571333 SRR13571334 SRR13571335 SRR13571336 SRR13571337 SRR13571338 SRR13571339 SRR13571340 SRR13571341 SRR13571342 SRR13571343 SRR13571344 SRR13571345 SRR13571346 SRR13571347 SRR13571348 SRR13571349 SRR13571350 SRR13571351 SRR13571352 SRR13571353 SRR13571354 SRR13571355 SRR13571356 SRR13571357 SRR13571358 SRR13571359 SRR13571360)
SRRID=${INPUT[$INDEX]}

SERIES_LIBRARY_LAYOUT=(SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE SINGLE)
SAMPLE_LIBRARY_LAYOUT=${SERIES_LIBRARY_LAYOUT[$INDEX]}

if [[ -e "$SRRID.tsv.gz" ]]; then
  exit
fi

prefetch --output-directory "/scratch/$USER" $SRRID | grep "\S" &> "$SRRID.log"

if [[ -d $SRRID ]]; then
  mv $SRRID/* "/scratch/$USER/$SRRID" && rm -rf $SRRID
fi

(cd "/scratch/$USER" && fasterq-dump --outdir $SRRID --temp $SRRID --threads 16 $SRRID) &>> "$SRRID.log"

if [[ $SAMPLE_LIBRARY_LAYOUT == "PAIRED" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*{1,2}.fastq" --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ $SAMPLE_LIBRARY_LAYOUT == "SINGLE" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*.fastq" --singleEnd --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ -n $(grep -l "Succeeded   : 13" "$SRRID.log") ]]; then
  gzip --stdout "/scratch/$USER/$SRRID/results/featureCounts/merged_gene_counts.txt" > "$SRRID.tsv.gz"
fi

rm -rf "/scratch/$USER/$SRRID"

