#!/bin/bash -l

#$ -e /dev/null
#$ -o /dev/null
#$ -P tuberculosis
#$ -pe omp 16
#$ -t 1-175

module load sratoolkit/2.10.5
module load nextflow/19.10.0

INDEX=$(($SGE_TASK_ID-1))

INPUT=(SRR6369992 SRR6369993 SRR6369994 SRR6369995 SRR6369996 SRR6369997 SRR6369998 SRR6369999 SRR6370000 SRR6370001 SRR6370002 SRR6370003 SRR6370004 SRR6370005 SRR6370006 SRR6370007 SRR6370008 SRR6370009 SRR6370010 SRR6370011 SRR6370012 SRR6370013 SRR6370014 SRR6370015 SRR6370016 SRR6370017 SRR6370018 SRR6370019 SRR6370020 SRR6370021 SRR6370022 SRR6370023 SRR6370024 SRR6370025 SRR6370026 SRR6370027 SRR6370028 SRR6370029 SRR6370030 SRR6370031 SRR6370032 SRR6370033 SRR6370034 SRR6370035 SRR6370036 SRR6370037 SRR6370038 SRR6370039 SRR6370040 SRR6370041 SRR6370042 SRR6370043 SRR6370044 SRR6370045 SRR6370046 SRR6370047 SRR6370048 SRR6370049 SRR6370050 SRR6370051 SRR6370052 SRR6370053 SRR6370054 SRR6370055 SRR6370056 SRR6370057 SRR6370058 SRR6370059 SRR6370060 SRR6370061 SRR6370062 SRR6370063 SRR6370064 SRR6370065 SRR6370066 SRR6370067 SRR6370068 SRR6370069 SRR6370070 SRR6370071 SRR6370072 SRR6370073 SRR6370074 SRR6370075 SRR6370076 SRR6370077 SRR6370078 SRR6370079 SRR6370080 SRR6370081 SRR6370082 SRR6370083 SRR6370084 SRR6370085 SRR6370086 SRR6370087 SRR6370088 SRR6370089 SRR6370090 SRR6370091 SRR6370092 SRR6370093 SRR6370094 SRR6370095 SRR6370096 SRR6370097 SRR6370098 SRR6370099 SRR6370100 SRR6370101 SRR6370102 SRR6370103 SRR6370104 SRR6370105 SRR6370106 SRR6370107 SRR6370108 SRR6370109 SRR6370110 SRR6370111 SRR6370112 SRR6370113 SRR6370114 SRR6370115 SRR6370116 SRR6370117 SRR6370118 SRR6370119 SRR6370120 SRR6370121 SRR6370122 SRR6370123 SRR6370124 SRR6370125 SRR6370126 SRR6370127 SRR6370128 SRR6370129 SRR6370130 SRR6370131 SRR6370132 SRR6370133 SRR6370134 SRR6370135 SRR6370136 SRR6370137 SRR6370138 SRR6370139 SRR6370140 SRR6370141 SRR6370142 SRR6370143 SRR6370144 SRR6370145 SRR6370146 SRR6370147 SRR6370148 SRR6370149 SRR6370150 SRR6370151 SRR6370152 SRR6370153 SRR6370154 SRR6370155 SRR6370156 SRR6370157 SRR6370158 SRR6370159 SRR6370160 SRR6370161 SRR6370162 SRR6370163 SRR6370164 SRR6370165 SRR6370166)
SRRID=${INPUT[$INDEX]}

SERIES_LIBRARY_LAYOUT=(PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED PAIRED)
SAMPLE_LIBRARY_LAYOUT=${SERIES_LIBRARY_LAYOUT[$INDEX]}

if [[ -e "$SRRID.tsv.gz" ]]; then
  exit
fi

prefetch --output-directory "/scratch/$USER" $SRRID | grep "\S" &> "$SRRID.log"
fasterq-dump --outdir "/scratch/$USER/$SRRID" --temp "/scratch/$USER/$SRRID" --threads 16 "/scratch/$USER/$SRRID" &>> "$SRRID.log"

if [[ $SAMPLE_LIBRARY_LAYOUT == "PAIRED" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*{1,2}.fastq" --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ $SAMPLE_LIBRARY_LAYOUT == "SINGLE" ]]; then
  nextflow run -profile singularity -revision 1.4.2 -work-dir "/scratch/$USER/$SRRID/work" nf-core/rnaseq --reads "/scratch/$USER/$SRRID/*.fastq" --singleEnd --genome GRCh38 --skipBiotypeQC --outdir "/scratch/$USER/$SRRID/results" &>> "$SRRID.log"
fi

if [[ -n $(grep -l "Succeeded   : 13" "$SRRID.log") ]]; then
  gzip --stdout "/scratch/$USER/$SRRID/results/featureCounts/merged_gene_counts.txt" > "$SRRID.tsv.gz"
fi

rm -rf "/scratch/$USER/$SRRID"

